etc/picongpu/jureca-jsc/batch.tpl

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+# Copyright 2013-2019 Axel Huebl, Richard Pausch, Rene Widera, Sergei Bastrakov
+#
+# This file is part of PIConGPU.
+#
+# PIConGPU is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PIConGPU is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with PIConGPU.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+# PIConGPU batch script for JURECA's SLURM batch system
+
+#SBATCH --account=!TBG_nameProject
+#SBATCH --partition=!TBG_queue
+#SBATCH --time=!TBG_wallTime
+# Sets batch job's name
+#SBATCH --job-name=!TBG_jobName
+#SBATCH --nodes=!TBG_nodes
+#SBATCH --ntasks=!TBG_tasks
+#SBATCH --ntasks-per-node=!TBG_devicesPerNode
+#SBATCH --mem=!TBG_memPerNode
+#SBATCH --mail-type=!TBG_mailSettings
+#SBATCH --mail-user=!TBG_mailAddress
+#SBATCH --workdir=!TBG_dstPath
+
+#SBATCH -o stdout
+#SBATCH -e stderr
+
+
+## calculations will be performed by tbg ##
+.TBG_queue="batch"
+
+# settings that can be controlled by environment variables before submit
+.TBG_mailSettings=${MY_MAILNOTIFY:-"NONE"}
+.TBG_mailAddress=${MY_MAIL:-"someone@example.com"}
+.TBG_author=${MY_NAME:+--author \"${MY_NAME}\"}
+.TBG_nameProject=${proj:-""}
+.TBG_profile=${PIC_PROFILE:-"~/picongpu.profile"}
+
+# number of available/hosted devices per node in the system
+.TBG_numHostedDevicesPerNode=2
+
+# required devices per node for the current job
+.TBG_devicesPerNode=`if [ $TBG_tasks -gt $TBG_numHostedDevicesPerNode ] ; then echo $TBG_numHostedDevicesPerNode; else echo $TBG_tasks; fi`
+
+# host memory per device
+.TBG_memPerCPU="$((126000 / $TBG_devicesPerNode))"
+# host memory per node
+.TBG_memPerNode="$((TBG_memPerCPU * TBG_devicesPerNode))"
+
+# number of cores to block per device - we got 24 cores (including hyper-threads) per device
+.TBG_coresPerCPU=24
```

Comment:
we can use this value for `export OMP_NUM_THREADS`

Reply:
Sure, but in this file it is only used once. Maybe it is better to define smth in the profile, since we have to use it [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/4e8a90ee247486e32ea4bc354669cd04391a6f97/etc/picongpu/jureca-jsc/batch_picongpu.profile.example#L89) as well?

Reply:
I would like to keep it in the tpl to allow also the usage with modules loaded by hand. But we can remove this definition and add only a comment above `export OMP_NUM_THREADS`.

You pointed to a very interesting point: **maybe** it would be much nicer for tbg to have a system description and a tpl which is only describing the batch system handling.

Reply:
Guess the latter is worth discussing, separate from this PR. Added to the [list](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2869#issuecomment-458311754).